types/index.d.ts

```diff
@@ -476,6 +498,46 @@ declare class Bone {
    * bone.remove(true)  // => DELETE FROM ... WHERE ...
    */
   remove(forceDelete?: boolean): number;
+
+  /**
+   * update or insert record.
+   * @example
+   * bone.upsert() // INERT ... VALUES ON DUPLICATE KEY UPDATE ...
+   * bone.upsert({ hooks: false })
+   * @param opts queryOptions
+   */
+  upsert(opts?: QueryOptions): Spell & Promise<number>;
+
+  /**
+   * update rows
+   * @param changes data changes
+   * @param opts query options
+   */
+  update(changes: Object, opts?: QueryOptions): Spell & Promise<number>;
+
+  /**
+   * create instance
+   * @param opts query options
+   */
+  create(opts?: QueryOptions): Spell & Promise<Bone>;
+
+  /**
+   * reload instance
+   */
+  reload(): Promise<Bone>;
+
+  /**
+   * delete row
+   * @param forceDelete force update
+   * @param opts query options
+   */
+  remove(forceDelete?: boolean, opts?: QueryOptions): Spell & Promise<number>;
```

Comment:
overloaded methods with more arguments should be put before the generic one. https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#function-overloads